scripts/compress_images.py

```
@@ -0,0 +1,247 @@
+
+                if new_size < original_size * self.tolerance:
+                    image_compressibility_dicts.append({
+                        'path': file_path,
+                        'original_size': original_size,
+                        'new_size': new_size
+                    })
+            else:
+                logging.error(
+                    '[ERROR]: %s occurred on file %s',
+                        file_path,
+                        result.stderr
+                )
+
+        return image_compressibility_dicts
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[CompressedImageInfo]. List of all compressible
+            images over repository with attributes as path, original_size
+            and new_size.
+        """
+        result_images: List[CompressedImageInfo] = []
+
+        for file_path in self.input_path.glob('**/*.*'):
+            file_extension = file_path.suffix.lower()
+            if file_extension not in self.ALL_IMAGES_EXTENSIONS:
+                continue
+
+            get_tempdir = tempfile.gettempdir()
+            temp_compressed = (
+                pathlib.Path(get_tempdir) / f'compressed_{file_path.name}'
+            )
+            result = self.run_compression_logic(
+                file_path,
+                temp_compressed,
+            )
+            if result:
+                result_images.extend(result)
+
+        return result_images
+
+    def compress_images(
+            self, result_images: List[CompressedImageInfo]
+        ) -> None:
+        """Compress the identified images.
+
+        Args:
+            result_images: List[CompressedImageInfo]. List of images
+                to be compressed.
+        """
+        # Remove existing output directory and create a new one.
+        if os.path.exists(self.output_dir):
```

You don't need this `if` check since errors are ignored by `shutil.rmtree`

---

Ahh, yes good catch. Removed it.